test/tests/functional/pbs_resv_confirm_hook.py

```diff
@@ -0,0 +1,403 @@
+
+        offset = 10
+        duration = 30
+        rid = self.server.submit_resv(offset, duration)
+
+        attrs = {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2')}
+        self.server.expect(RESV, attrs, id=rid)
+
+        self.server.delete(rid)
+        msg = 'Hook;Server@%s;Reservation ID - %s' % \
+              (self.server.shortname, rid)
+        self.server.log_match(msg, tail=True, interval=1, max_attempts=10)
+
+    @tags('hooks')
+    def test_delete_degraded_resv(self):
+        """
+        Testcase to submit and confirm an advance reservation, turn the mom
+        off, delete the degraded reservation and verify the resv_confirm
+        hook ran the correct number of times.
+        """
+        self.server.import_hook(self.hook_name,
+                                TestResvConfirmHook.advance_resv_hook_script)
+
+        offset = 10
+        duration = 30
+        rid = self.server.submit_resv(offset, duration)
+
+        attrs = {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2')}
+        self.server.expect(RESV, attrs, id=rid)
+        msg = 'Hook;Server@%s;Reservation ID - %s' % (self.server.shortname,
+                                                      rid)
+        self.server.log_match(msg, tail=True, interval=1, max_attempts=10)
+
+        self.mom.stop()
+
+        attrs['reserve_state'] = (MATCH_RE, 'RESV_DEGRADED|10')
+        self.server.expect(RESV, attrs, id=rid)
+
+        self.server.delete(rid)
+        msg = 'Hook;Server@%s;Reservation ID - %s' % (self.server.shortname,
+                                                      rid)
+        self.server.log_match(msg, tail=True, interval=1, max_attempts=10)
+
+    @tags('hooks')
+    def test_server_down_case_1(self):
+        """
+        Testcase to submit and confirm an advance reservation, turn the server
+        off, turn the server on, delete the reservation after start and verify
+        the resvconfirm hook ran.
+        """
+        self.server.import_hook(self.hook_name,
+                                TestResvConfirmHook.advance_resv_hook_script)
+
+        offset = 10
+        duration = 300
+        rid = self.server.submit_resv(offset, duration)
+
+        attrs = {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2')}
+        self.server.expect(RESV, attrs, id=rid)
+
+        self.server.stop()
+
+        self.server.start()
+
+        time.sleep(11)
+
+        self.server.delete(rid)
+
+        msg = 'Hook;Server@%s;Reservation ID - %s' % \
+              (self.server.shortname, rid)
+        self.server.log_match(msg, tail=True, interval=1, max_attempts=10)
+
+    @tags('hooks')
+    @timeout(300)
+    def test_server_down_case_2(self):
+        """
+        Testcase to submit and confirm an advance reservation, turn the
+        server off, wait for the reservation duration to finish, turn the
+        server on and verify the resvconfirm hook never ran.
+        """
```

Comment:
I'm not sure what I was originally thinking, but I see your point.  If the server's down the submission would fail and the scheduler down case is later, which is the actual interesting one.  I'll drop this case as well.

Reply:
Case dropped in  00d1d03.